extensions/wikia/ArticleAsJson/ArticleAsJson.class.php

```diff
@@ -322,4 +450,57 @@ private static function isIconSize( $sizeParam ) {
 	private static function isInfoIcon( $templateType ) {
 		return $templateType == TemplateClassificationService::TEMPLATE_INFOICON;
 	}
+
+	/**
+	 * @param $originalHeight
+	 * @param $originalWidth
+	 *
+	 * @return array
+	 */
+	private static function scaleIconSize( $originalHeight, $originalWidth ) {
+		$height = $originalHeight;
+		$width = $originalWidth;
+		$maxHeight = self::ICON_SCALE_TO_MAX_HEIGHT;
+
+		if ( $originalHeight > $maxHeight ) {
+			$height = $maxHeight;
+			$width = intval( $maxHeight * $originalWidth / $originalHeight );
+		}
+
+		return [
+			'height' => $height,
+			'width' => $width
+		];
+	}
+
+	private static function getMediaDetailWithSizeFallback(
+		$title, $mediaDetailConfig, $fallbackSize=self::MAX_MERCURY_IMAGE_WIDTH
+	) {
+		$mediaDetail = WikiaFileHelper::getMediaDetail( $title, $mediaDetailConfig );
+		if ( $mediaDetail['width'] == 0 ) {
+			$mediaDetail['width'] = $fallbackSize;
+
+			\Wikia\Logger\WikiaLogger::instance()->error(
+				'Media width was empty - fallback to fallbackSize',
```

Comment:
How about considering https://docs.google.com/document/d/1rGZ0we9E0iNFu2Y5SO89m3mCK2uALvvCwEUQQ_OS8qE and using more meaningful message here?

E.g. "ArticleAsJson - Media width was empty, using fallback size". Also keys should be underscored, so `media_details` and `fallback_size`.